services/src/main/scala/cromwell/services/metadata/impl/MetadataServiceActor.scala

```diff
@@ -49,9 +49,17 @@ case class MetadataServiceActor(serviceConfig: Config, globalConfig: Config, ser
 
   private val metadataReadTimeout: Duration =
     serviceConfig.getOrElse[Duration]("metadata-read-query-timeout", Duration.Inf)
+  private val metadataReadRowNumberSafetyThreshold: Int =
+    serviceConfig.getOrElse[Int]("metadata-read-row-number-safety-threshold", 1000000)
```

Comment:
TOL perhaps the subject of another ticket, but are we confident this is a good default (readers will not explode below this limit, reasonable metadata requests will not be denied) and/or are we going to override it to something else in the Terra config?

Reply:
I actually don't know. We need some statistic. I made current choice based on a single workflow `c3b9d7b0-3fc6-4f25-9097-2526cfadb9ca`, mentioned in BA-6423, which killed Cromwell in the past and which has ~16000000 rows. So I just took a number "much lower, but not too low, to prevent returning failures for good workflows"